docs/book/src/developer/architecture/controllers/tuning.md

```diff
@@ -0,0 +1,122 @@
+
+So then the next step after looking at rate limiting metrics, is to look at the "work queue depth" panel in the [Controller-Runtime dashboard](http://localhost:3001/d/abe29aa7-e44a-4eef-9474-970f95f08ee6/controller-runtime?orgId=1).
+
+Assuming that one controller is struggling with its own work queue, the next step is to look at why this is happening. It might be that the average duration of each reconcile is high for some reason. This can be checked in the "Reconcile Duration by Controller" panel in the [Controller-Runtime dashboard](http://localhost:3001/d/abe29aa7-e44a-4eef-9474-970f95f08ee6/controller-runtime?orgId=1).
+
+If this is the case, then it is time to start looking at traces, looking for the longer spans in average (or total). Unfortunately traces are not implemented in Cluster API, so alternative approaches must be used, like looking at condition transitions or at logs to figure it out what the slowest operations are.
+
+And so on.
+
+Please note that there are also cases where CAPI controllers are just idle waiting for something else to happen on the infrastructure side. In this case investigating bottlenecks requires access to a different set of metrics. Similar considerations apply if the issue is slowness of the API server or of the network.
+
+## Runtime tuning options
+
+Cluster API offers a set of options that can be set on the controller deployment at runtime, without the need of changing the CAPI code.
+
+- Client-go rate limiting; by increasing the client-go rate limits we allow a controller to make more API server calls per second (`--kube-api-qps`) or to have a bigger burst to handle spikes (`--kube-api-burst`). Please note that these settings must be increased carefully, because being too aggressive on the API server might lead to different kind of problems.
+
+- Controller concurrency (e.g. via `--kubeadmcontrolplane-concurrency`); by increasing the number of concurrent reconcile loops for each controller  it is possible to help the system in keeping the work queue clean, and thus in being more fast in reconciling the desired state. Also in this case, trade-offs should be considered, because by increasing concurrency not only the controller footprint is going to increase, but also the number of API server calls is likely going to increase (see previous point).
+
+- Resync period (`--sync-period`); this setting defines the interval after which reconcile events for all current objects will be triggered. Historically this value in Cluster API is much lower than the default in controller runtime (10m vs. 24h). This has some advantages, because e.g. it is a fallback in case controller struggle to pick up events from external infrastructure. But it also has impact at scale when a controller gets a sudden spike of events at every resync period. This can be mitigated by increasing the resync period.
```

Comment:
```suggestion
- Resync period (`--sync-period`); this setting defines the interval after which reconcile events for all current objects will be triggered. Historically this value in Cluster API is much lower than the default in controller runtime (10m vs. 10h). This has some advantages, because e.g. it is a fallback in case controller struggle to pick up events from external infrastructure. But it also has impact at scale when a controller gets a sudden spike of events at every resync period. This can be mitigated by increasing the resync period.
```
Didn't double-check before